### ISSUE TITLE
salt-ssh: using cmd.run inside a state

### ISSUE THREAD

**Issue by mostafahussein** [saltstack/salt#41623]
Hello,

When i try to use `cmd.run` within a state and call it using salt-ssh, the result of cmd.run is the following:
```
{'retcode': 0, '_error': 'Failed to return clean data', 'stderr': "'{'run': 'shell'}.run' is not available.", 'stdout': ''}
```
For example add this inside a state `{% set epoch = salt.cmd.run('date +%s') %}`

Note: I am using cmd.run for other cases which works correctly with a regular minion but not with salt-ssh

the value of `epoch` will be as above .
Is this an issue ? Is there a work around in case i needed `cmd.run` the execution module inside a state ?

**Comment by gtmanfred** [comment]
I am able to replicate this.

```
[root@salt ~]# cat /srv/salt/test.sls
{% set epoch = salt.cmd.run('date +%s') %}
/tmp/gttest:
  file.managed:
    - contents: {{epoch}}
[root@salt ~]# salt-ssh \* state.apply test
[WARNING ] /root/src/salt/salt/utils/templates.py:80: DeprecationWarning: Starting in 2015.5, cmd.run uses python_shell=False by default, which doesn't support shellisms (pipes, env variables, etc). cmd.run is currently aliased to cmd.shell to prevent breakage. Please switch to cmd.shell or set python_shell=True to avoid breakage in the future, when this aliasing is removed.

self:
----------
          ID: /tmp/gttest
    Function: file.managed
      Result: True
     Comment: File /tmp/gttest updated
     Started: 16:10:47.923765
    Duration: 18.96 ms
     Changes:
              ----------
              diff:
                  New file

Summary for self
------------
Succeeded: 1 (changed=1)
Failed:    0
------------
Total states run:     1
Total run time:  18.960 ms
[root@salt ~]# cat /tmp/gttest
{'stdout': '', '_error': 'Failed to return clean data', 'stderr': "'{'run': 'shell'}.run' is not available.", 'retcode': 0}
[root@salt ~]# salt-ssh --versions-report
Salt Version:
           Salt: 2016.11.5

Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: Not Installed
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.9.6
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.8
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.5 (default, Nov  6 2016, 00:28:07)
   python-gnupg: Not Installed
         PyYAML: 3.12
          PyZMQ: 16.0.2
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.5.1
            ZMQ: 4.1.6

System Versions:
           dist: centos 7.3.1611 Core
        machine: x86_64
        release: 3.10.0-514.16.1.el7.x86_64
         system: Linux
        version: CentOS Linux 7.3.1611 Core
```

But it looks like it is not a regression and has been happening since before 2016.3.0

```
[root@salt salt]# cat /srv/salt/test.sls
{% set epoch = salt.cmd.run('date +%s') %}
/tmp/gttest:
  file.managed:
    - contents: {{epoch}}
[root@salt salt]# salt-ssh \* state.apply test
self:
----------
          ID: /tmp/gttest
    Function: file.managed
      Result: True
     Comment: File /tmp/gttest updated
     Started: 16:13:30.600745
    Duration: 70.362 ms
     Changes:
              ----------
              diff:
                  New file

Summary for self
------------
Succeeded: 1 (changed=1)
Failed:    0
------------
Total states run:     1
Total run time:  70.362 ms
[root@salt salt]# cat /tmp/gttest
1496852008
```

It looks like this has been fixed and implemented in 2017.5 which we are working to get an RC out soon for.

Thanks,
Daniel

**Comment by drawsmcgraw** [comment]
Adding my +1, for `cmd.script` as well.

```
----------
          ID: push_watches
    Function: cmd.script
        Name: salt://es5/files/push_watches.sh
      Result: False
     Comment: State 'cmd.script' was not found in SLS 'es5.push_watcher_watches'
              Reason: 'cmd.script' is not available.
     Changes:   
```


```
# salt-ssh --versions-report
Salt Version:
           Salt: 2016.11.4
 
Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 2.6.0
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.7.2
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.8
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: 3.4.3
         pygit2: Not Installed
         Python: 2.7.5 (default, Nov  6 2016, 00:28:07)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 15.3.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.1.4
 
System Versions:
           dist: centos 7.3.1611 Core
        machine: x86_64
        release: 3.10.0-514.16.1.el7.x86_64
         system: Linux
        version: CentOS Linux 7.3.1611 Core
```
